Week 04/Lecture 08/Assignment 02/employee_manager/src/main/java/com/example/employee_manager/dao/EmployeeDAO.java

```diff
@@ -0,0 +1,62 @@
+package com.example.employee_manager.dao;
+
+import com.example.employee_manager.model.Employee;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+@Repository
+public class EmployeeDAO {
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
```

Comment:
Refer Assignment 2, which ways of DI is better. Should not use field injection.

Reply:
Thankyou for the feedback, i have resolved that, i change the injection in the some class into constructor based injection, you can see in this latest commit, thankyou.

[improvement commit](https://github.com/affandyfandy/java-alif/pull/8/commits/0cefdda545fb69fdd519d7d956de10bbd614a169)
[improvement commit-2](https://github.com/affandyfandy/java-alif/pull/8/commits/bd14f51ddbcdbcb6b609a0615235387ac4c26b9f)